### PR TITLE
Added proper responses to nts messages.

### PR DESCRIPTION
**Pull request by davidv1992**

### CODE REVIEW

File: ntp-proto/src/packet/mod.rs [hunk 1]
```diff
@@ -456,7 +456,87 @@ impl<'a> NtpPacket<'a> {
...
+                        .iter()
+                        .chain(input.efdata.encrypted.iter())
+                        .filter_map(|f| {
+                            if let ExtensionField::NtsCookiePlaceholder { cookie_length } = f {
+                                let new_cookie = keyset.encode_cookie(cookie);
+                                if new_cookie.len() > *cookie_length as usize {
+                                    None
+                                } else {
+                                    Some(ExtensionField::NtsCookie(Cow::Owned(new_cookie)))
+                                }
```
Comment (folkertdev) [inline ntp-proto/src/packet/mod.rs:524]: this logic can be factored out (and hence deduplicated)

File: ntp-proto/src/packet/mod.rs [hunk 2]
```diff
@@ -456,7 +456,87 @@ impl<'a> NtpPacket<'a> {
...
+                    authenticated: input
+                        .efdata
+                        .authenticated
+                        .iter()
+                        .filter_map(|f| {
+                            if matches!(f, ExtensionField::UniqueIdentifier(_)) {
+                                Some(f.clone())
+                            } else {
+                                None
+                            }
```
Comment (folkertdev) [inline ntp-proto/src/packet/mod.rs:509]: can use `into_iter` here if the `authenticated` field is moved down

File: ntp-proto/src/packet/mod.rs [hunk 3]
```diff
@@ -456,7 +456,87 @@ impl<'a> NtpPacket<'a> {
...
+                        .untrusted
+                        .iter()
+                        .chain(input.efdata.authenticated.iter())
+                        .filter_map(|f| {
+                            if matches!(f, ExtensionField::UniqueIdentifier(_)) {
+                                Some(f.clone())
+                            } else {
+                                None
+                            }
+                        })
```
Comment (folkertdev) [inline ntp-proto/src/packet/mod.rs:474]: this operation is done a bunch of times. Maybe factor this out into a function?

File: ntp-proto/src/packet/mod.rs (hunk 1 above)
Comment (folkertdev) [reply ntp-proto/src/packet/mod.rs:524]: maybe even move this whole closure into its own function

File: ntp-proto/src/packet/mod.rs (hunk 3 above)
Comment (davidv1992) [reply ntp-proto/src/packet/mod.rs]: done

File: ntp-proto/src/packet/mod.rs (hunk 2 above)
Comment (davidv1992) [reply ntp-proto/src/packet/mod.rs]: done

File: ntp-proto/src/packet/mod.rs (hunk 1 above)
Comment (davidv1992) [reply ntp-proto/src/packet/mod.rs:519]: done

**Review by folkertdev — APPROVED** [review]

**Pull request merged**